marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/models/generateModelConfig.sjs

```diff
@@ -0,0 +1,62 @@
+/*
+  Copyright (c) 2020 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+const consts = require("/data-hub/5/impl/consts.sjs");
+const config = require("/com.marklogic.hub/config.sjs");
+const hent = require("/data-hub/5/impl/hub-entities.xqy");
+
+
+const result = {};
+const protectedPaths = [];
```

Comment:
There's no need to define these ahead of time. Just assign properties directly to "result", makes for less code - i.e. declare "result" as:

```
result = {
  protectedPaths: [],
  queryRolesets: [],
  index-config: {},
  options: {}
};
```

That also has a nice benefit of immediately telling the reader what the response object looks like.

Reply:
Makes sense. Updated in here b6777af